src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,422 @@
+
+    /**
+     * The status of the guild scheduled event
+     * @type {GuildScheduledEventStatus}
+     */
+    this.status = GuildScheduledEventStatuses[data.status];
+
+    /**
+     * The type of hosting entity associated with the scheduled event
+     * @type {GuildScheduledEventEntityType}
+     */
+    this.entityType = GuildScheduledEventEntityTypes[data.entity_type];
+
+    if ('entity_id' in data) {
+      /**
+       * The id of the hosting entity associated with the scheduled event
+       * @type {?Snowflake}
+       */
+      this.entityId = data.entity_id;
+    } else {
+      this.entityId ??= null;
+    }
+
+    if ('user_count' in data) {
+      /**
+       * The number of users who are subscribed to this guild scheduled event
+       * @type {?number}
+       */
+      this.userCount = data.user_count;
+    } else {
+      this.userCount ??= null;
+    }
+
+    if ('creator' in data) {
+      /**
+       * The user that created this guild scheduled event
+       * @type {?User}
+       */
+      this.creator = this.client.users._add(data.creator);
+    } else {
+      this.creator ??= this.client.users.resolve(this.creatorId);
+    }
+
+    /* eslint-disable max-len */
+    /**
+     * Represents the additional metadata for a {@link GuildScheduledEvent}
+     * @see {@link https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-guild-scheduled-event-entity-metadata}
+     * @typedef {Object} GuildScheduledEventEntityMetadata
+     * @property {?string} location The location of the guild scheduled event
+     */
+    /* eslint-enable max-len */
+
+    if ('entity_metadata' in data) {
+      if (data.entity_metadata) {
+        /**
+         * Additional metadata
+         * @type {?GuildScheduledEventEntityMetadata}
+         */
+        this.entityMetadata = {
+          location: data.entity_metadata.location ?? this.entityMetadata?.location ?? null,
+        };
+      } else {
+        this.entityMetadata = null;
+      }
+    } else {
+      this.entityMetadata ??= null;
+    }
+  }
+
+  /**
+   * The timestamp the guild scheduled event was created at
+   * @type {number}
+   * @readonly
+   */
+  get createdTimestamp() {
+    return SnowflakeUtil.timestampFrom(this.id);
+  }
+
+  /**
+   * The time the guild scheduled event was created at
+   * @type {Date}
+   * @readonly
+   */
+  get createdAt() {
+    return new Date(this.createdTimestamp);
+  }
+
+  /**
+   * The time the guild scheduled event will start at
+   * @type {Date}
+   * @readonly
+   */
+  get scheduledStartAt() {
+    return new Date(this.scheduledStartTimestamp);
+  }
+
+  /**
+   * The time the guild scheduled event will end at,
+   * or `null` if the event does not have a scheduled time to end
+   * @type {?Date}
+   * @readonly
+   */
+  get scheduledEndAt() {
+    return this.scheduledEndTimestamp ? new Date(this.scheduledEndTimestamp) : null;
```

Comment:
Following the new semantics we're using in latest PRs:
```suggestion
    return this.scheduledEndTimestamp && new Date(this.scheduledEndTimestamp);
```